chrome_windows.go

```diff
@@ -1,39 +1,64 @@
 package kooky
 
-/*
-#cgo LDFLAGS: -lCrypt32
-#define NOMINMAX
-#include <windows.h>
-#include <Wincrypt.h>
-char* decrypt(byte* in, int len, int *outLen) {
-	DATA_BLOB input, output;
-	LPWSTR pDescrOut =  NULL;
-	input.cbData = len;
-	input.pbData = in;
-	CryptUnprotectData(
-		&input,
-		&pDescrOut,
-		NULL,
-		NULL,
-		NULL,
-		0,
-		&output);
-	*outLen = output.cbData;
-	return output.pbData;
+// https://groups.google.com/d/msg/golang-nuts/bUetmxErnTw/GHC5obCcmTMJ
+// https://play.golang.org/p/fknP9AuLU-
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+const (
+	CRYPTPROTECT_UI_FORBIDDEN = 0x1
+)
+
+var (
+	dllcrypt32  = syscall.NewLazyDLL("Crypt32.dll")
+	dllkernel32 = syscall.NewLazyDLL("Kernel32.dll")
+
+	procDecryptData = dllcrypt32.NewProc("CryptUnprotectData")
+	procLocalFree   = dllkernel32.NewProc("LocalFree")
+)
+
+type data_blob struct {
+	cbData uint32
+	pbData *byte
 }
-void doFree(char* ptr) {
-	free(ptr);
+
+func newBlob(d []byte) *data_blob {
+	if len(d) == 0 {
+		return &data_blob{}
+	}
+	return &data_blob{
+		pbData: &d[0],
+		cbData: uint32(len(d)),
+	}
 }
-*/
-import "C"
 
-func setChromeKeychainPassword(password []byte) []byte {
-	return password
+func (b *data_blob) toByteArray() []byte {
+	d := make([]byte, b.cbData)
+	copy(d, (*[1 << 30]byte)(unsafe.Pointer(b.pbData))[:])
```

Comment:
Hmmm. I wonder if it wouldn't be better to declare (mostly dummy) structs to pull the data into, rather than random numeric pointer offsets. Then again, ¯\\\_(ツ)\_/¯  :-)